pkg/vmconfig/crypto/crypto_reconciler_pre.go

```diff
@@ -546,6 +550,12 @@ func getCryptoKeyFromEncryptionClass(
 		return cryptoKey{}, ErrInvalidKeyProvider
 	}
 
+	if obj.Spec.KeyID != "" {
+		if ok, _ := m.IsValidKey(ctx, obj.Spec.KeyProvider, obj.Spec.KeyID); !ok {
```

Comment:
Note that this method of key validation is not supported with the Native key provider. The Native key provider does not support directly generating keys either. So it seems specifying a `keyID` with native `keyProvider` is not a valid use-case? If it is considered a valid use-case, will need to update this to skip calling `IsValidKey` if the provider is native.

Reply:
Earlier in development, I tracked if the provider was native. We can reintroduce that for this.

Reply:
Earlier in development, I tracked if the provider was native. We can reintroduce that for this.